de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/wizard/CsvTypeReviewPage.java

```diff
@@ -0,0 +1,330 @@
+	 * Composite)
+	 */
+	@Override
+	public void createControl(Composite parent) {
+
+		Composite content = new Composite(parent, SWT.NONE);
+		content.setLayout(new GridLayout());
+		content.setLayoutData(new GridData(GridData.FILL_BOTH | GridData.GRAB_HORIZONTAL | GridData.GRAB_HORIZONTAL));
+		setControl(content);
+
+		Composite typeNameComposite = new Composite(content, SWT.NONE);
+		typeNameComposite.setLayout(new GridLayout(2, false));
+		GridData data = new GridData(GridData.HORIZONTAL_ALIGN_FILL | GridData.GRAB_HORIZONTAL);
+		typeNameComposite.setLayoutData(data);
+
+		Label label = new Label(typeNameComposite, SWT.NONE);
+		label.setText(TYPE_NAME_LABEL);
+
+		typeNameText = new Text(typeNameComposite, SWT.SINGLE | SWT.BORDER);
+		GridData dataText = new GridData(GridData.HORIZONTAL_ALIGN_FILL | GridData.GRAB_HORIZONTAL);
+		data.horizontalAlignment = SWT.END;
+		data.widthHint = WITH_TEXT;
+		typeNameText.setLayoutData(dataText);
+		typeNameText.addSelectionListener(this);
+		typeNameText.addModifyListener(this);
+
+		table = new Table(content, SWT.CHECK | SWT.BORDER | SWT.H_SCROLL);
+		table.setLayout(new GridLayout());
+		table.setLayoutData(new GridData(GridData.FILL_BOTH | GridData.GRAB_HORIZONTAL | GridData.GRAB_VERTICAL));
+		table.setHeaderVisible(true);
+		table.addSelectionListener(this);
+
+		TableColumn columnColumnName = new TableColumn(table, SWT.NULL);
+		columnColumnName.setWidth(COLUMN_NAME_WIDTH);
+		columnColumnName.setText(COLUMN_NAME_LABEL);
+
+		TableColumn columnType = new TableColumn(table, SWT.NONE | SWT.DROP_DOWN);
+		columnType.setText(COLUMN_TYPE_LABEL);
+		columnType.setWidth(COLUMN_TYPE_WIDTH);
+
+	}
+
+	/**
+	 * Sets the input for the mapping table
+	 * 
+	 * @param columnNames
+	 *            a list of columns in the file
+	 * @param requirementType
+	 *            an existing requirement type
+	 */
+	public void setInput(List<String> columnNames, RequirementType requirementType) {
+		this.requirementType = requirementType;
+		this.typeNameText.setText(requirementType.getName());
+		isNewType = requirementType.getTypeInstance().eContainer() == null;
+		if (!isNewType) {
+			typeNameText.setEditable(false);
+		} else {
+			typeNameText.setEditable(true);
+		}
+
+		table.clearAll();
+
+		for (String column : columnNames) {
+			Integer index = columnNames.indexOf(column);
+			TableItem item;
+			if (tableItems.size() <= index) {
+				item = new TableItem(table, SWT.NULL);
+				tableItems.add(item);
+			} else {
+				item = tableItems.get(index);
+			}
+			item.setText(0, column);
+
+			if (isNewType) {
+				item.setText(TYPE_COLUMN_INDEX, requirementType.getAttributes().get(index).getType());
+				item.setChecked(true);
+				createDropDownEditor(item, ATTRIBUTE_TYPES);
+			} else {
+				if (index < requirementType.getAttributes().size()) {
+					item.setText(TYPE_COLUMN_INDEX,
+							requirementType.getName() + TYPE_LABEL_REQUIREMENT_ATTRIBUTE_SEPARATOR

```

Comment:
Maybe we could encapsulate this into a small method that takes requirementType and attribute to give back the separated text?